### PR TITLE
Prefix complex delegation certificate option

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Prefix complex delegation certificate option
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  - breaking       # the API has changed in a breaking way
  # - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] The change log section in the PR description has been filled in
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - round trip tests
  - integration tests
  See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] The changelog section in the PR is updated to describe the change
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_delegation-certificate.cli [hunk 1]
```diff
@@ -33,6 +33,22 @@ Available options:
...
   --stake-address ADDRESS  Target stake address (bech32 format).
+  --combined-drep-verification-key STRING
+                           DRep verification key (Bech32 or hex-encoded).
+  --combined-drep-verification-key-file FILE
+                           Filepath of the DRep verification key.
+  --combined-drep-key-hash HASH
+                           DRep verification key hash (either Bech32-encoded or
+                           hex-encoded). Zero or more occurences of this option
+                           is allowed.
+  --combined-stake-pool-verification-key STRING
```
Comment (carbolymer) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_delegation-certificate.cli:44]: what's the difference between combined-stake-pool and stake-pool keys?

**Review by carbolymer — CHANGES_REQUESTED** [review]
I don't think this is the solution agreed upon. To my knowledge we are supposed to remove combined certs support: https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1693317040694109?thread_ts=1693267316.476929&cid=C031GFLC3DF
cc: @CarlosLopezDeLara

**Comment by CarlosLopezDeLara** [comment]
> I don't think this is the solution agreed upon. To my knowledge we are supposed to remove combined certs support: https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1693317040694109?thread_ts=1693267316.476929&cid=C031GFLC3DF cc: @CarlosLopezDeLara

My bad, I just became aware that this kind of StakeANDVote delegation certificate existed in ledger. So if it exists in Ledger we will support it.

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_delegation-certificate.cli (hunk 1 above)
Comment (CarlosLopezDeLara) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_delegation-certificate.cli:44]: It looks like it is to make it clear that this is for a combined certificate, isn't it?

**Comment by CarlosLopezDeLara** [comment]
@newhoggy 

Still feels a little overcomplicated, tho.  I wonder if it would be better split in 2 delegation commands, something like:
```
cardano-cli conway governance drep delegation-certificate 
                ( --stake-verification-key STRING
                | --stake-verification-key-file FILE
                | --stake-script-file FILE
                | --stake-address ADDRESS
                )
                ( --drep-verification-key STRING
                | --drep-verification-key-file FILE
                | --drep-key-hash HASH
                | --drep-script-hash HASH
                | --always-abstain
                | --always-no-confidence
                )
                --out-file FILE
```
and
```
cardano-cli conway governance drep combined-delegation-certificate 
                         ( --stake-verification-key STRING
                         | --stake-verification-key-file FILE
                         | --stake-script-file FILE
                         | --stake-address ADDRESS
                         )
                         ( --drep-verification-key STRING
                         | --drep-verification-key-file FILE
                         | --drep-key-hash HASH
                         | --always-abstain
                         | --always-no-confidence
                         )
                         ( --stake-pool-verification-key STRING
                         | --cold-verification-key-file FILE
                         | --stake-pool-id STAKE_POOL_ID
                         )
                         --out-file FILE
```
which again draws me to the idea that all of this should eventually really live under  `cardano-cli stake-address` where current stake delegation lives. Yes, this then brings the problem of having drep delegation option in Babbage. We need to determine what will end up under era top-level commands.

**Review by carbolymer — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**